examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,522 @@
+# events occuring with a constant rate in a given time interval (``Exposure``).
+# Here we model the frequency ``y = ClaimNb / Exposure``,
+# which is still a (scaled) Poisson distribution.
+#
+# A very important property of the Poisson distribution is its mean-variance
+# relation: The variance is proportional to the mean.
+
+df_train, df_test, X_train, X_test = train_test_split(df, X, random_state=2)
+
+# Some of the features are colinear, we use a weak penalization to avoid
+# numerical issues.
+glm_freq = PoissonRegressor(alpha=1e-2)
+glm_freq.fit(X_train, df_train.Frequency, sample_weight=df_train.Exposure)
+
+
+def mean_deviance(estimator, y, y_pred, weights):
+    if hasattr(estimator, "_family_instance"):
+        return estimator._family_instance.deviance(y, y_pred, weights) / len(y)
+    else:
+        return np.nan
+
+
+def score_estimator(
+    estimator, X_train, X_test, df_train, df_test, target, weights
+):
+    res = []
+
+    for subset_label, X, df in [
+        ("train", X_train, df_train),
+        ("test", X_test, df_test),
+    ]:
+        y, _weights = df[target], df[weights]
+
+        for score_label, metric in [
+            ("D² explained", None),
+            ("mean deviance", partial(mean_deviance, estimator)),
+            ("mean abs. error", mean_absolute_error),
+            ("mean squared error", mean_squared_error),
+        ]:
+            if estimator.__class__.__name__ == "ClaimProdEstimator":
+                # ClaimProdEstimator is the product of frequency and severity
+                # models, denormalized by the exposure values.
+                # It does not fully follow the scikit-learn API and we
+                # must handle it separately.
+                y_pred = estimator.predict(X, exposure=df.Exposure.values)
+            else:
+                y_pred = estimator.predict(X)
+            if metric is None:
+                if not hasattr(estimator, "score"):
+                    continue
+                score = estimator.score(X, y, _weights)
+            else:
+                score = metric(y, y_pred, _weights)
+
+            res.append(
+                {"subset": subset_label, "metric": score_label, "score": score}
+            )
+
+    res = (
+        pd.DataFrame(res)
+        .set_index(["metric", "subset"])
+        .score.unstack(-1)
+        .round(3)
+    )
+    return res
+
+
+scores = score_estimator(
+    glm_freq,
+    X_train,
+    X_test,
+    df_train,
+    df_test,
+    target="Frequency",
+    weights="Exposure",
+)
+print(scores)
+
+##############################################################################
+#
+# We can visually compare observed and predicted values, aggregated by
+# the drivers age (``DrivAge``), vehicle age (``VehAge``) and the insurance
+# bonus/malus (``BonusMalus``).
+
+fig, ax = plt.subplots(2, 2, figsize=(16, 8))
+fig.subplots_adjust(hspace=0.3, wspace=0.2)
+
+plot_obs_pred(
```

Comment:
These plots are really neat, but what is the takeaway here? A short analysis would be helpful.

Reply:
Additional description was added (a while ago) below.